src/main/java/pl/put/poznan/json/tools/model/decorations/NullRemover.java

```diff
@@ -4,14 +4,46 @@
 import pl.put.poznan.json.tools.model.JsonObject;
 import pl.put.poznan.json.tools.service.WrongInputException;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class NullRemover extends JsonDecorator {
 
     public NullRemover(JsonObject jsonObject) {
         super(jsonObject);
     }
 
     public String getJson() throws WrongInputException {
-        // jsonObject.getJson() zwraca String który musisz przetworzyć odpowiednio i zwrócić
-        return jsonObject.getJson();
+        String jsonString                               = jsonObject.getJson();
+        jsonString                                      = jsonString.substring(1, jsonString.length() - 1);
+        List<String> jsonAttributes                     = Arrays.asList(jsonString.split("\\s*,\\s*"));
+        ArrayList<List<String>> jsonAttributesSplited   = new ArrayList<>();
+        ArrayList<String> jsonAttributesRemovedNulls    = new ArrayList<>();
+
+        splitAttributes(jsonAttributes, jsonAttributesSplited);
+        removeNulls(jsonAttributesSplited, jsonAttributesRemovedNulls);
+
+        String jsonFinal = "{\n\t" + String.join(",\n\t", jsonAttributesRemovedNulls) + "\n}";
```

Comment:
ten poziom tez nie pasuje za bardzo. Na upartego wrzuć to w rerurna bezpośrednio, ale wolałbym metodę prywatną w stylu joinJson()